README
NAME
    Perl::Critic::More - Supplemental policies for Perl::Critic

SYNOPSIS
      perl -MCPAN -e'install Perl::Critic::More'
      perlcritic -theme more lib/Foo.pm

AFFILIATION
    This file has no functionality, but instead is a placeholder for a loose
    collection of Perl::Critic policies. All of those policies will have an
    "Affiliation" section announcing their participation in this grouping.

DESCRIPTION
    This is a collection of Perl::Critic policies that are not included in
    the Perl::Critic core for a variety of reasons:

    *   Experimental

        Some policies need some time to work out their kinks, test
        usability, or gauge community interest. A subset of these will end
        up in the core Perl::Critic someday.

    *   Requires special dependencies

        For example, some policies require development versions of PPI (or
        some other CPAN module). These will likely end up in the
        Perl::Critic core when their dependencies are fulfilled.

    *   Peripheral to Perl

        For example, the "Editor::RequireEmacsFileVariables" policy is
        metacode. These are not part of Perl::Critic's mission.

    *   Special purpose

        For example, policies like "CodeLayout::RequireASCII" designed to
        scratch itches not felt by most of the community. These will always
        remain in a Perl::Critic supplement instead of in the core.

    All of these policies have the theme "more" so they can be turned off as
    a group via .perlcriticrc by adding this line:

      theme = not more

    The special purpose ones may be part of the "notrecommended" theme.
    Avoid these via:

      theme = not notrecommended

    Sorry about the double-negative... See "CONFIGURATION" in Perl::Critic
    for details on how to interact with themes.

SEE ALSO
    Perl::Critic

    Perl::Critic::Bangs - Andy Lester's fantastic list of code pet peeves

    Perl::Critic::Lax - RJBS' more-lenient versions of some core
    Perl::Critic policies

    parrot - the parrot team has developed a few specialized Perl::Critic
    policies of their own

AUTHOR
    Chris Dolan <cdolan@cpan.org>

    The included policies may have other authors -- please see them
    individually.

    This distribution is controlled by the Perl::Critic team. If you want to
    add a policy to this collection, check out our Subversion repository and
    mailing lists at <http://perlcritic.tigris.org>.

COPYRIGHT
    Copyright (c) 2006-2008 Chris Dolan

    This program is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself. The full text of this license can
    be found in the LICENSE file included with this module.



Changes
##############################################################################
#      $URL: http://perlcritic.tigris.org/svn/perlcritic/trunk/distributions/Perl-Critic-More/Changes $
#     $Date: 2013-10-25 17:08:59 -0700 (Fri, 25 Oct 2013) $
#   $Author: thaljef $
# $Revision: 4211 $
##############################################################################

[1.002] Released on 2013-10-25


    New Policies:
    * ValuesAndExpressions::RequireConstantOnLeftSideOfEquality -- In
      case you accidentally say "if ($foo = 42) {...}"

    "use 5.006001" in Makefile.PL and Build.PL.

[1.001] Released on 2013-10-25

    This release was a flop.  Do not use it!
    Changes summarized above under version 1.002

[1.000] Released on 2008-05-26

     Bug Fix:
     * Modules::PerlMinimumVersion now handles $PERL_VERSION being an object,
       as it is in perl 5.10.

     Policy moved:
     * ValuesAndExpressions::ProhibitMagicNumbers has been moved into the
       core Perl::Critic distribution.

     Dependencies:
     * Now requires Perl::Critic 1.082.

[0.999_002] Released on 2008-05-18
[0.999_001] Released on 2008-05-04

     Changes summarized into 1.000 above.  For exact details, see Changes in
     0.999_002 on BackPAN.

[0.16] Released on 2007-08-12

     Bugfix release
      * PPI version back down to 1.118, but turn off ProhibitMagicNumbers with older PPI
      * work around v-string warning
     [Thanks to users of CPAN::Reporter for identifying bugs!]

[0.15]  Released on 2007-08-10

     Bugfix release
      * Readonly dependency omitted
      * PPI version => 1.199_03

[0.14]  Released on 2007-08-09

     New Policies:
     * ErrorHandling::RequireUseOfExceptions
     * ValuesAndExpressions::RestrictLongStrings
     * ValuesAndExpressions::ProhibitMagicNumbers

[0.12]  Released on 2006-11-23

     New Policies:
     * CodeLayout::RequireASCII
     * Modules::RequirePerlVersion
     * Modules::PerlMinimumVersion
       (depends on CPAN module Perl::MinimumVersion > 0.13)

[0.11]  Released on 2006-11-15
     Upload to the right PAUSE account this time...

[0.1]   Released on 2006-11-15

     Create this new distribution by moving stuff from the core Perl::Critic

     New Policies:
     * Editor::RequireEmacsFileVariables


Build.PL
#!perl

#      $URL: http://perlcritic.tigris.org/svn/perlcritic/trunk/Perl-Critic-More/Build.PL $
#     $Date: 2008-05-18 18:40:27 -0500 (Sun, 18 May 2008) $
#   $Author: clonezone $
# $Revision: 2366 $

use 5.006;

use strict;
use warnings;

use English qw< -no_match_vars >;

use Module::Build;


my $class = Module::Build->subclass( code => <<'END_SUBCLASS' );

    sub ACTION_test {
        my ($self) = @_;

        $self->depends_on('manifest');

        return $self->SUPER::ACTION_test();
    }

    sub ACTION_authortest {
        my ($self) = @_;

        $self->depends_on('build');
        $self->depends_on('manifest');
        $self->depends_on('distmeta');

        $self->test_files( qw< t xt/author > );
        $self->recursive_test_files(1);

        $self->depends_on('test');

        return;
    } # end ACTION_authortest()

    sub ACTION_distdir {
        my ($self) = @_;

        $self->depends_on('authortest');

        return $self->SUPER::ACTION_distdir();
    } # end ACTION_distdir

END_SUBCLASS

my $builder = $class->new(
    module_name         => 'Perl::Critic::More',
    dist_author         => 'Chris Dolan <cdolan@cpan.org>',
    license             => 'perl',
    dynamic_config      => 1,

    requires            => {
        'Perl::Critic'  => 1.082,
        'Readonly'      => 1.03,
    },

    build_requires      => {
        'Test::More'    => 0,
    },

    recommends          => {
        'Perl::MinimumVersion' => '0.14',
    },

    create_packlist     => 1,

    sign                => 0,

    create_readme       => 1,

    create_makefile_pl  => 'traditional',

    meta_merge          => {
        resources   => {
            homepage    => 'http://perlcritic.com',
            license     => 'http://dev.perl.org/licenses/',
            Repository  => 'http://perlcritic.tigris.org/svn/perlcritic/trunk/Perl-Critic-More/',
            MailingList => 'http://perlcritic.tigris.org/servlets/SummarizeList?listName=users',
        }
    },

    add_to_cleanup      => [
        qw<
            MANIFEST.bak
            Makefile
            Makefile.old
            pod2htm?.tmp
            pm_to_blib
            Perl-Critic-More-*
        >,
    ],
);


$builder->create_build_script();

##############################################################################
# Local Variables:
#   mode: cperl
#   cperl-indent-level: 4
#   fill-column: 78
#   indent-tabs-mode: nil
#   c-indentation-style: bsd
# End:
# ex: set ts=8 sts=4 sw=4 tw=78 ft=perl expandtab shiftround :
